sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -178,13 +195,105 @@ internal class ViewAnnotationManagerImpl(
 
     if (viewAnnotationOptions.isEmpty()) return null
     val coordinates = coordinatesFromAnnotations(viewAnnotationOptions)
-    val paddings = calculateEdgeInsets(viewAnnotationOptions, edgeInsets)
-    return mapboxMap.cameraForCoordinates(
+    var cameraOptionForCoordinates = mapboxMap.cameraForCoordinates(
       coordinates,
-      paddings,
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
       bearing,
       pitch
     )
+
+    var isCorrectBound = false
+    // keep north, east, west and southmost Annotation and its frame to adjust paddings for zoom.
+    var north: Pair<ViewAnnotationOptions, Rect?>? = null
+    var east: Pair<ViewAnnotationOptions, Rect?>? = null
+    var west: Pair<ViewAnnotationOptions, Rect?>? = null
+    var south: Pair<ViewAnnotationOptions, Rect?>? = null
+
+    // we run the loop twice to adjust bounds correctly to fit all the annotations.
```

Comment:
Nit - reword to hint that there is still no guarantee that bounds will fit all the annotations?